public/app/features/explore/RawPrometheus/RawPrometheusContainer.tsx

```diff
@@ -133,13 +132,14 @@ export class RawPrometheusContainer extends PureComponent<Props, PrometheusConta
       (frame: DataFrame | undefined): frame is DataFrame => !!frame && frame.length !== 0
     );
 
+    const title = this.state.resultsStyle === TABLE_RESULTS_STYLE.raw ? 'Raw' : 'Table';
     const label = this.state?.resultsStyle !== undefined ? this.renderLabel() : 'Table';
 
     // Render table as default if resultsStyle is not set.
     const renderTable = !this.state?.resultsStyle || this.state?.resultsStyle === TABLE_RESULTS_STYLE.table;
 
     return (
-      <Collapse label={label} loading={loading} isOpen>
+      <PanelChrome title={title} actions={label} loadingState={loading ? LoadingState.Loading : LoadingState.Done}>
```

Comment:
i wonder if we can simply pass `loadingState` like we do in other viz from `Explore.tsx` and use it here, i think the logic in https://github.com/grafana/grafana/pull/78084/files#diff-d1e0351d2a892577c267481e278db65aeec5a0f543f8531c4f0ca353c7c1dd34R36 was used just because the Collapse component only supported a boolean loading state, while PanelChrome can receive the actual `LoadingState` used everywhere else.

@ifrost @gtk-grafana you know better than me here, am i missing something?

Reply:
Yes, think we should be able just use the loadingState from `queryResponse.state` 🤔

Reply:
I've switched to using `queryResponse.state`, but I'm running into a weird issue with a failing test when removing `const loadingInState = selectIsWaitingForData(exploreId)(state);` I've removed `loadingInState` and I'm just using `queryResponse.state` for the `loading prop`. The weird thing is that the test will pass even if loadingInState is just initialized and not being used anywhere in the component. I've looked at the `selectIsWaitingForData`s implementation and this function is not modifying any state but simply checking for the `queryResponse` and returning true or false based on the `LoadingState` type. 

Steps to reproduce:
1. Run the `RawPrometheusContainer.test.tsx`
2. The test will fail
3. Add this to `mapStateToProps` function: `const loadingInState = selectIsWaitingForData(exploreId)(state);` 
4. The test passes

Reply:
Long story short: TemplateSrv needs to be mocked for this test. If it's not mocked you get the error `Error: Uncaught [TypeError: Cannot read properties of undefined (reading 'replace')]` at [applyOverrides](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/explore/RawPrometheus/RawPrometheusContainer.tsx#L123).

Why does it work with `selectIsWaitingForData(exploreId)(state);`? When you import (yes, just import) `selectIsWaitingForData` it initializes TemplateSrv... 

I think it's a great example of how tightly coupled our code is and why global singletons are confusing.

- RawPrometheus/RawPrometheusContainer.tsx imports `import { selectIsWaitingForData } from '../state/query';` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/explore/RawPrometheus/RawPrometheusContainer.tsx#L15))
- ...which `import { runRequest } from '../../query/state/runRequest';` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/explore/state/query.ts#L53
))

- ...which `import { emitDataRequestEvent } from './queryAnalytics';` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/query/state/runRequest.ts#L29))

- ...which `import { getDashboardSrv } from '../../dashboard/services/DashboardSrv';` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/query/state/queryAnalytics.ts#L4))

- ...which `import { saveDashboard } from 'app/features/manage-dashboards/state/actions';` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/dashboard/services/DashboardSrv.ts#L9))

- ...which `import { getLibraryPanel } from '../../library-panels/state/api';` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/manage-dashboards/state/actions.ts#L15))

- ...which `import { DashboardModel } from 'app/features/dashboard/state';` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/library-panels/state/api.ts#L4))

- ...which `import { onTimeRangeUpdated } from 'app/features/variables/state/actions';` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/dashboard/state/DashboardModel.ts#L28))

- ...which `import { getTemplateSrv, TemplateSrv } from '../../templating/template_srv';` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/variables/state/actions.ts#L27))

- ...which when is imported sets the default TemplateSrv `setTemplateSrv(srv);` ([code](https://github.com/grafana/grafana/blob/v10.2.1/public/app/features/templating/template_srv.ts#L386))

So just importing `import { selectIsWaitingForData } from '../state/query';` poor `RawPrometheusContainer.tsx` imports a bunch of dashboard related code and ends up initializing the TemplateSrv.

Reply:
trading a side effect for another, can't we just do 
```ts
beforeAll(() => {
  getTemplateSrv();
});
```

in the test file? it should probably be the test setup responsibility to set up the environment for the test

Reply:
ah lol, i misread your comment, you were suggesting the same thing 😵‍💫

Reply:
@ifrost thank you for drilling through this and finding the root cause!

![image](https://media2.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif)